awswrangler/s3/_write_dataset.py

```diff
@@ -137,24 +172,51 @@ def _to_dataset(
     use_threads: bool,
     mode: str,
     partition_cols: Optional[List[str]],
+    partitions_types: Optional[Dict[str, str]],
+    catalog_id: Optional[str],
+    database: Optional[str],
+    table: Optional[str],
+    table_type: Optional[str],
+    transaction_id: Optional[str],
     bucketing_info: Optional[Tuple[List[str], int]],
     boto3_session: boto3.Session,
     **func_kwargs: Any,
 ) -> Tuple[List[str], Dict[str, List[str]]]:
     path_root = path_root if path_root.endswith("/") else f"{path_root}/"
 
+    commit_trans: bool = False
+    if table_type == "GOVERNED":
+        # Check whether to skip committing the transaction (i.e. multiple read/write operations)
+        if transaction_id is None:
+            _logger.debug("`transaction_id` not specified, beginning transaction")
+            transaction_id = begin_transaction(read_only=False, boto3_session=boto3_session)
+            commit_trans = True
+
```

Comment:
Block required to handle cases where a `transaction_id` is created **outside** the `wr.s3.to_parquet` call (i.e. anticipating multiple read and writes before committing the transaction)